src/extraction/FStar.Extraction.ML.Modul.fs

```diff
@@ -983,25 +983,83 @@ let rec extract_sig (g:env_t) (se:sigelt) : env_t * list<mlmodule1> =
         | Sig_let (lbs, _) ->
           let attrs = se.sigattrs in
           let quals = se.sigquals in
-          let attrs, post_tau =
+          let maybe_postprocess_lbs lbs =
+            let post_tau =
               match U.extract_attr' PC.postprocess_extr_with attrs with
-              | None -> attrs, None
-              | Some (ats, (tau, None)::_) -> ats, Some tau
-              | Some (ats, args) ->
-                  Errors.log_issue se.sigrng (Errors.Warning_UnrecognizedAttribute,
-                                         ("Ill-formed application of `postprocess_for_extraction_with`"));
-                  attrs, None
-          in
-          let postprocess_lb (tau:term) (lb:letbinding) : letbinding =
-              let lbdef = Env.postprocess (tcenv_of_uenv g) tau lb.lbtyp lb.lbdef in
+              | None -> None
+              | Some (_, (tau, None)::_) -> Some tau
+              | Some _ ->
+                  Errors.log_issue
+                      se.sigrng
+                      (Errors.Warning_UnrecognizedAttribute,
+                       "Ill-formed application of 'postprocess_for_extraction_with'");
+                  None
+            in
+            let postprocess_lb (tau:term) (lb:letbinding) : letbinding =
+              let env = tcenv_of_uenv g in
+              let lbdef = 
+                Profiling.profile
+                     (fun () -> Env.postprocess env tau lb.lbtyp lb.lbdef)
+                     (Some (Ident.string_of_lid (Env.current_module env)))
+                     "FStar.Extraction.ML.Module.post_process_for_extraction"
+              in
               { lb with lbdef = lbdef }
+            in
+            match post_tau with
+            | None -> lbs
+            | Some tau -> fst lbs, List.map (postprocess_lb tau) (snd lbs)
           in
-          let lbs = (fst lbs,
-                      (match post_tau with
-                       | Some tau -> List.map (postprocess_lb tau) (snd lbs)
-                       | None -> (snd lbs)))
+          let maybe_normalize_for_extraction lbs = 
+            let norm_steps =
+              match U.extract_attr' PC.normalize_for_extraction_lid attrs with
+              | None -> None
+              | Some (_, (steps, None)::_) ->
+                let steps = 
+                  //just normalizing the steps themselves, so that the user
+                  //does not have to write a literal at every use of the attribute
+                  N.normalize 
+                    [Env.UnfoldUntil delta_constant; Env.Zeta; Env.Iota; Env.Primops]
+                    (tcenv_of_uenv g)
+                    steps
+                in
+                begin
+                match Cfg.try_unembed_simple (EMB.e_list EMB.e_norm_step) steps with
```

Comment:
Calling unembed here for the steps required me to refactor the handling of embeddings, which was previously buried within the normalizer.